src/uucore/src/lib/features/format/num_format.rs

```diff
@@ -272,10 +280,30 @@ impl Formatter for Float {
             format_float_non_finite(x, self.case)
         };
 
+        let flag = if x >= 0. {
```

Comment:
seems duplicated code with above, could you please have a look to create a function ?
thanks

Reply:
I created 2 functions and considered combining them using traits, but that seems unnecessary. Let me know if there's any other way to improve.